### PR TITLE
Fix race condition in reactive add

### PR DESCRIPTION
**Pull request by gcollin**
Where sometimes the list is run before the add is taking place.

### CODE REVIEW

**Comment by gsmet** [comment]
@cescoffier can you have a look at this one?

File: using-mongodb-client/src/main/java/org/acme/rest/json/ReactiveFruitResource.java [hunk 1]
```diff
@@ -22,7 +22,6 @@
 
     @POST
     public CompletionStage<List<Fruit>>  add(Fruit fruit) {
-        fruitService.add(fruit);
-        return list();
+        return fruitService.add(fruit).thenCompose(aVoid -> list());
```
Comment (cescoffier) [inline using-mongodb-client/src/main/java/org/acme/rest/json/ReactiveFruitResource.java]: I would use `x` instead of `aVoid`.

Comment (gcollin) [reply using-mongodb-client/src/main/java/org/acme/rest/json/ReactiveFruitResource.java]: Sure.

Comment (gcollin) [reply using-mongodb-client/src/main/java/org/acme/rest/json/ReactiveFruitResource.java]: done

**Pull request merged**